src/notation/internal/notationstyle.cpp

```diff
@@ -89,3 +89,53 @@ Notification NotationStyle::styleChanged() const
 {
     return m_styleChanged;
 }
+
+mu::io::path NotationStyle::selectStyleFile(bool forLoad)
+{
+    mu::io::path dir = configuration()->userStylesPath();
+    QString filter = qtrc("notation", "MuseScore Styles") + " (*.mss)";
+    return forLoad
+           ? interactive()->selectOpeningFile(qtrc("notation", "Load Style"), dir, filter)
+           : interactive()->selectSavingFile(qtrc("notation", "Save Style"), dir, filter);
+}
+
+void NotationStyle::loadStyle()
+{
+    mu::io::path stylePath = selectStyleFile(true);
+    if (stylePath.empty()) {
+        return;
+    }
+
+    QFile f(stylePath.toQString());
+    if (!f.open(QIODevice::ReadOnly) || !Ms::MStyle::isValid(&f)) {
+        interactive()->error(trc("notation", "The style file could not be loaded."),
+                             f.errorString().toStdString(), { IInteractive::Button::Ok },
+                             IInteractive::Button::Ok, IInteractive::Option::WithIcon);
+        return;
+    }
+
+    m_undoStack->prepareChanges();
+    if (!m_getScore->score()->loadStyle(stylePath.toQString()) && interactive()->warning(
+            trc("notation",
+                "Since this style file is from a different version of MuseScore, your score is not guaranteed to display correctly."),
+            trc("notation", "Click OK to load anyway."), { IInteractive::Button::Ok, IInteractive::Button::Cancel },
+            IInteractive::Button::Ok, IInteractive::Option::WithIcon).standardButton()
+        == IInteractive::Button::Ok) {
+        m_getScore->score()->loadStyle(stylePath.toQString(), true);
+    }
+    m_undoStack->commitChanges();
```

Comment:
This is not a good place for this code.
We adhere to the MVVM architectural pattern
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel
User interaction must be at the ViewModel level
This place is Model, there should be no user interaction, ideally. Otherwise, this code (loading, saving styles) cannot be used in other cases, for example, during auto testing, or on the backend, or anywhere else.

Reply:
ViewModel classes are in `modulename/view`

Reply:
It was Roman's suggestion but I agree it's not the best place to put it, I had it in NotationActionController originally. Thankfully these are about the only two actions I've implemented recently (out of more than 60!) that do require user interaction with dialogs etc. But we need a discussion about how this is going to work with the plugin engine etc.